waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/metastore/ReconnectingMetaStoreClientFactory.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (C) 2016-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hotels.bdp.waggledance.metastore;
+
+import java.lang.reflect.Proxy;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.hotels.hcommon.hive.metastore.MetaStoreClientException;
+import com.hotels.hcommon.hive.metastore.client.api.CloseableMetaStoreClient;
+import com.hotels.hcommon.hive.metastore.client.api.MetaStoreClientFactory;
+
+class ReconnectingMetaStoreClientFactory implements MetaStoreClientFactory {
```

Comment:
Where do these "ReconnectingXxx" classes come from? I thought this PR was going to be just modifying a few classes to use new common lib for wherever there was overlap as opposed to adding new classes.

Reply:
Things had to change so it can work for both projects, also a lot of these reflection MetaStoreClient classes didn't have tests, good time to add them IMO

Reply:
Also name clashes, Circus Train and Waggle Dance classes have the same names that weren't descriptive, this is a MetaStoreClient that reconnects if it is disconnected -> ReconnectingMetaStoreClient